### ISSUE TITLE
1-24-link containers test does not work in HaaS

### ISSUE THREAD

**Issue by mhagen-vmware** [vmware/vic#7401]
This part of the test consistently keeps failing:
```
    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run --net jedi --link first:1st debian ping -c1 1st
    Should Be Equal As Integers  ${rc}  0
    Should Not Contain  ${output}  Error
```

The biggest difference it looks like to me from running this locally and from HaaS is that HaaS is blocking DNS queries to 8.8.8.8.... and it looks like from the portlayer logs that it fails right around the step that it does a DNS query but add the internal DNS servers to the install script did not change anything and other test steps appear to make the DNS query step just fine as well. This is the first step in the test that uses network alias, I am wondering if we have a special DNS path through our code when container aliases are being used?

[1-24-Docker-Link-VCH-0-1797-container-logs.zip](https://github.com/vmware/vic/files/1760285/1-24-Docker-Link-VCH-0-1797-container-logs.zip)

**Comment by hmahmood** [comment]
Yes we do have an a dns server running in the port layer. I will look through the logs.

**Comment by mhagen-vmware** [comment]
This might be resolved, waiting on more CI builds to get better idea.

**Comment by hickeng** [comment]
EndpointVM is getting the following:
```
Feb 26 2018 20:03:21.119Z DEBUG writing "nameserver 10.142.7.21\n" to /.tether/etc/resolv.conf
Feb 26 2018 20:03:21.120Z DEBUG writing "nameserver 10.142.7.22\n" to /.tether/etc/resolv.conf
Feb 26 2018 20:03:21.121Z DEBUG writing "nameserver 10.166.17.90\n" to /.tether/etc/resolv.conf
```
and the containers are being correctly configured to talk to the portlayer for proxied DNS resolution.
```
Feb 26 2018 20:04:26.188Z DEBUG writing "nameserver 172.17.0.1\n" to /.tether/etc/resolv.conf
```

**Comment by mhagen-vmware** [comment]
yea, that was when I manually configured vic-machine with --dns-servers, still didn't help as it would seem.  The latest build failed on the the other link container test still:
https://ci-vic.vmware.com/vmware/vic/17057/8

**Comment by mhagen-vmware** [comment]
but it actually made it through 1-24, so this is consistently passing now for some reason.. I am going to close this then and open a new issue for the other linked container failures in the other tests.
